mount/mount.go

```diff
@@ -189,6 +189,15 @@ func (d *Datastore) Put(key ds.Key, value []byte) error {
 	return cds.Put(k, value)
 }
 
+// Sync implements Datastore.Sync
+func (d *Datastore) Sync(prefix ds.Key) error {
+	cds, _, k := d.lookup(prefix)
+	if cds == nil {
+		return ErrNoMount
+	}
+	return cds.Sync(k)
```

Comment:
We need to sync _all_ datastores under the prefix. This will only sync the one responsible for the given key.

If we have two mounts `["/", "/blocks"]`, `Sync("/")` will currently _just_ sync "/".

Reply:
Maybe we can modify the test datastore to record all `Sync` calls then use that o make sure sync is getting called correctly?

Reply:
ah cool, I didn't realize we wanted mount to be able to do that. I figured that if we wanted submounts that the "/foo" would point to another mount datastore that was "/foo/bar".

should be fixable though. will take a look later today

Reply:
Yeah, `lookup` is just for finding the datastore where we should read/write a key.